etc/tests/test_bootstrapper.py

```diff
@@ -163,6 +164,183 @@ def test_main_method_with_dir_outputs(monkeypatch, s3_setup, tmpdir):
+        'some_property': 'some property value',
+        'outputs': [
+            {
+                'source': 'gs://project/bucket/file.md',
+                'type': 'markdown'
+            }
+        ]
+    }
+
+    with tmpdir.as_cwd():
+        with open(input_metadata_file, 'w') as f:
+            json.dump(custom_metadata, f)
+
+    main_method_setup_execution(monkeypatch, s3_setup, tmpdir, argument_dict)
+    # /tmp/mlpipeline-ui-metadata.json should now have been updated
+
+    try:
+        with open(output_metadata_file, 'r') as f:
+            metadata = json.load(f)
+            assert metadata.get('some_property') is not None
+            assert metadata['some_property'] == custom_metadata['some_property']
+            assert metadata.get('outputs') is not None
+            assert isinstance(metadata['outputs'], list)
+            assert len(metadata['outputs']) == 2
+            for output in metadata['outputs']:
+                if output.get('storage') is not None:
+                    assert output['storage'] == 'inline'
+                    assert output['type'] == 'markdown'
+                    assert '{}/{}/{}'.format(argument_dict['cos-endpoint'],
+                                             argument_dict['cos-bucket'],
+                                             argument_dict['cos-directory']) \
+                        in output['source']
+                    assert argument_dict['cos-dependencies-archive']\
+                        in output['source']
+                else:
+                    assert output['type'] ==\
+                        custom_metadata['outputs'][0]['type']
+                    assert output['source'] ==\
+                        custom_metadata['outputs'][0]['source']
+    except AssertionError:
+        raise
+    except Exception as ex:
+        # Potential reasons for failures:
+        # file not found, invalid JSON
+        print('Validation of "{}" failed: {}'.format(str(ex), ex))
+        assert False
+
+
+def test_process_metrics_method_invalid_metadata_file(monkeypatch, s3_setup, tmpdir):
+    """Test for process_metrics
+
+    Verifies that process_metrics produces a valid KFP UI metadata file if
+    the node's script | notebook generate an invalid metadata file, which
+    cannot be merged and is therefore overwritten.
+    """
+    argument_dict = {'cos-endpoint': 'http://' + MINIO_HOST_PORT,
+                     'cos-bucket': 'test-bucket',
+                     'cos-directory': 'test-directory',
+                     'cos-dependencies-archive': 'test-archive.tgz',
+                     'filepath': 'etc/tests/resources/test-notebookA.ipynb',
+                     'inputs': 'test-file.txt;test,file.txt',
+                     'outputs': 'test-file/test-file-copy.txt;test-file/test,file/test,file-copy.txt',
+                     'user-volume-path': None}
+
+    input_metadata_file = 'mlpipeline-ui-metadata.json'
+    output_metadata_file = '/tmp/mlpipeline-ui-metadata.json'
```

Comment:
ditto

Reply:
The test will now be skipped if the environment doesn't meet the file system prerequisites.